### ISSUE TITLE
Unwanted bindgen artifacts in WASM file.

### ISSUE THREAD

**Issue by zie1ony** [risc0/risc0#299]
# Problem
My goal is running risc0 verifier inside the WASM VM.

Here is my example:

`risc0-zkvm` included in dependencies:
```toml
[dependencies]
risc0-zkvm = { version = "1.0.0-rc.2", features = [ "verify-only" ] }
```

`verifier.rs` is my source code. It's enough to `use risc0_zkvm`, no need to interact with any code like `Recipt`:
```rust
#![no_main]

use risc0_zkvm;

extern "C" {
    pub fn load_data() -> u8;
}

#[no_mangle]
unsafe fn verify() {
    let _ = load_data();
}
```

See full code here: https://github.com/zie1ony/risc0-wasm-unwanted-bindgen

Compile it to wasm:
```bash
$ cargo build --release --target wasm32-unknown-unknown
```

Now I can inspect it using `wasm2wat` from [WABT](https://github.com/WebAssembly/wabt) package.
```bash
$ wasm2wat target/wasm32-unknown-unknown/release/verify.wasm
(module
  (type (;0;) (func (param i32 i32)))
 ...
  (type (;21;) (func (param i32 i32 f64 i32 i32)))
  (import "env" "load_data" (func $load_data (type 11)))
  (import "__wbindgen_placeholder__" "__wbindgen_describe" (func $_ZN12wasm_bindgen19__wbindgen_describe17h402830b0880612dcE (type 12)))
  (import "__wbindgen_placeholder__" "__wbindgen_throw" (func $_ZN12wasm_bindgen16__wbindgen_throw17hc7b411560c64c5a5E (type 0)))
  (import "__wbindgen_externref_xform__" "__wbindgen_externref_table_grow" (func $_ZN12wasm_bindgen9externref31__wbindgen_externref_table_grow17h5d8fced3e7917d66E (type 13)))
  (import "__wbindgen_externref_xform__" "__wbindgen_externref_table_set_null" (func $_ZN12wasm_bindgen9externref35__wbindgen_externref_table_set_null17h644009d4f22d60bcE (type 12)))
  (func $verify (type 14)
    call $load_data
```

Those  `__wbindgen_placeholder__` imports, are not expected, and the wasm file has much more of `bindgen` artifacts.

```bash
$` wasm2wat target/wasm32-unknown-unknown/release/verify.wasm | grep bindgen | wc -l
9544
```

Of course WASM hosts outside the browser or Node will not be able to satisfy those imports and will fail.

# Reason
The reason is `getrandom` crate in the dependencies of `risc0-zkvm`. It comes from `rand`'s dependencies. `getrandom` devs supports `wasm32-unknown-unknown` target only when the `js` feature is on, but this expects the wasm file to be used in JS environment, which is really naive. And to support JS they use... `wasm-bindgen`. You can read more in the source code: https://github.com/rust-random/getrandom/blob/master/src/lib.rs#L60

# Solutions
I see three solutions:
1. Drop the `random` from dependencies for `verify-only` feature.
2. Remove `bindgen` artifacts somehow from the wasm file. There are WASM tools like [wasm-bindgen](https://rustwasm.github.io/wasm-bindgen/reference/cli.html) or WABT, but I haven't found any way of doing so.
3. Improve `getrandom` and make PR to their repo.

My preferable solution is the first one, as it reduces dependecies and improves the `rist0` repo. I might be wrong, but the random values are not needed for verification, right?

**Comment by zie1ony** [comment]
Wow, that was quick :) I checked and it worked.

**Comment by zie1ony** [comment]
Wow, that was quick :) I checked and it worked.

**Comment by flaub** [comment]
I'm going to leave this open until the PR lands, but thanks for checking!

**Comment by flaub** [comment]
Fixed in #302
